### PR TITLE
Migration changes

### PR DESCRIPTION
**Pull request by disq**
Document migration changes, to go with

https://github.com/cloudquery/cq-provider-sdk/pull/146
https://github.com/cloudquery/cq-provider-aws/pull/401
https://github.com/cloudquery/cloudquery/pull/406
https://github.com/cloudquery/cq-provider-template/pull/12

### CODE REVIEW

File: docs/developers/sdk/migrations/overview.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Every CloudQuery provider should ship with initial migration files to create tables.
+Before reading this section make sure you have read the [developing new provider](../../developing-new-provider.md) guide first.
+
+Since CloudQuery is backed by a SQL database, providers consistently change between versions and can create breaking changes. Migrations is a classic approach to support changes without rebuilding the database. Under the hood CloudQuery uses [go-migrate](https://github.com/golang-migrate/migrate).
+
+## Adding provider migrations support 
+
+The CloudQuery SDK allows to easily send a list of embedded SQL migration files from a provider to the core, we are required to only add the following to our provider struct:
+
+```go
```
Comment (roneli) [inline docs/developers/sdk/migrations/overview.md:12]: nit: maybe make this snippet workable

File: docs/developers/sdk/migrations/migration-tool.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func main() {
+	if err := migration.Run(context.Background(), resources.Provider(), ""); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
+		os.Exit(1)
+	}
+}
+```
+
+Tool options are:
```
Comment (roneli) [inline docs/developers/sdk/migrations/migration-tool.md:25]: I think the tool updated a little so we should reflect this in the docs

**Review by roneli — CHANGES_REQUESTED** [review]
Some changes, I think docs need a little update to match the actual final release version

File: docs/developers/sdk/migrations/overview.md (hunk 1 above)
Comment (disq) [reply docs/developers/sdk/migrations/overview.md:12]: By workable you mean? Also most of this is the original contents of `docs/developers/sdk/migrations.md` moved here

File: docs/developers/sdk/migrations/migration-tool.md (hunk 2 above)
Comment (disq) [reply docs/developers/sdk/migrations/migration-tool.md:25]: `-schema` added, that's all

File: docs/developers/sdk/migrations/overview.md (hunk 1 above)
Comment (roneli) [reply docs/developers/sdk/migrations/overview.md:12]: Copying it makes a vaild piece of go code, I think just wrap it with a function

**Review by roneli — APPROVED** [review]

**Pull request merged**